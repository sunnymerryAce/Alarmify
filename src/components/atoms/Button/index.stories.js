import React from 'react';
import '../../../App.css';
import Button from './index';

export default {
  title: 'Button',
};

export const Default = () => (
  <Button onClick={() => {}} text="Insert  Some Text" />
);
