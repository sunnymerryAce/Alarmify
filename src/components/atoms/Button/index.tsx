import React from 'react';
import styled from 'styled-components';

interface Props {
  onClick: (props: any) => void;
  text: string;
}

const Button: React.FC<Props> = (props) => {
  return (
    <StyledButton className="Button" onClick={props.onClick}>
      {props.text}
    </StyledButton>
  );
};

export default Button;

const StyledButton = styled.div`
  background-color: #1db954;
  font-size: 20px;
  font-weight: bold;
  line-height: 30px;
  border-radius: 50px;
  width: 70%;
  padding: 10px 0;
  margin: 7vh auto;
`;
