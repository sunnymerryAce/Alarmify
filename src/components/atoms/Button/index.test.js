import React from 'react';
import Button from './index';
import { shallow } from 'enzyme';

describe('<Button />', () => {
  const onClickFunction = () => {
    return true;
  };
  const text = 'TEST';

  const container = shallow(<Button onClick={onClickFunction} text={text} />);
  test('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });

  test('should have proper props', () => {
    expect(container.find('.Button').props()).toMatchObject({
      onClick: onClickFunction,
    });
  });

  test('should have proper text', () => {
    expect(container.find('.Button').text()).toBe(text);
  });
});
