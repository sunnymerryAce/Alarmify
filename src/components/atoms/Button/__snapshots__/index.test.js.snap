// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match the snapshot 1`] = `"<div class=\\"sc-bdVaJa lkGCXV Button\\">TEST</div>"`;
